### PR TITLE
Fix: #7. Add jsonschema

### PR DESCRIPTION
**Pull request by ioggstream**
## This PR

- adds application/json+schema application/json-instance+schema

## Notes

- there are normative statements inherited from  https://github.com/json-schema-org/json-schema-spec/blob/707431358d529dc8892610e4d8593cfdd1f090a3/jsonschema-core.xml

@relequestual @jasondesrosiers

### CODE REVIEW

**Comment by Relequestual** [comment]
Wondering if it might be worth not doing anything YAML related in a PR titled "Add JSON Schema".
Might be worth doing YAML related things in a separate PR? =]

File: draft-ietf-httpapi-rest-api-mediatypes.md [hunk 1]
```diff
@@ -252,6 +263,116 @@ Author:  See Authors' Addresses section
...
+
+Fragment identifier considerations:  same as {{schema-json}}
+
+Additional information:
+
+  Deprecated alias names for this type:  n/a
+
+  Magic number(s):  n/a
+
+  File extension(s):  json, schema.json
```
Comment (Relequestual) [inline draft-ietf-httpapi-rest-api-mediatypes.md:362]: The instance is specifically not a schema.

```suggestion
  File extension(s):  json
```

**Comment by jdesrosiers** [comment]
I'm not sure we need to register `application/schema-instance+json`. I'll raise the discussion on the JSON Schema Slack and see if anyone thinks it's worth keeping.

**Comment by ioggstream** [comment]
@jdesrosiers thanks, I just inherited it from the current jschema draft.

**Comment by jdesrosiers** [comment]
We discussed this on Friday and here's what we decided.

**YAML**
@Relequestual's concern about yaml was adding an unnecessary complication to JSON Schema, but we talked about how it would be limited to a YAML subset that is compatible with JSON (just like with Open API and JSON-LD). It's not YAML Schema, it's JSON Schema encoded as YAML. Since the point of writing a JSON Schema in YAML is for human readability, it's not clear why it would be necessary to transport a JSON Schema encoded in YAML. Storing it locally in YAML and developing in YAML makes sense, but transporting in YAML doesn't makes sense. @Relequestual is going to reach out to the community to see if an `application/schema+yaml` media-type is something people actually want.

**The `schema` media-type parameter**
I'll try to keep a long story short. Originally, the `schema` parameter had only one URI. Recently, it was changed to allow a comma separated list of URIs. That wasn't well enough thought out and we ended up removing it, but it accidentally got left in the IANA Considerations sections. What we want to register is the original one-URI version of `schema`. I'll suggest the necessary changes on the PR some time this week.

**The `application/instance+schema` media type**
We decided we do want to include this media type.

**Comment by ioggstream** [comment]
Hi @jdesrosiers 

> **YAML** @Relequestual's concern about yaml was adding an unnecessary complication to JSON Schema, but we talked about how it would be limited to a YAML subset that is compatible with JSON (just like with Open API and JSON-LD). 

Exactly

> It's not YAML Schema, it's JSON Schema encoded as YAML. 

Exactly.

> Since the point of writing a JSON Schema in YAML is for human readability, it's not clear why it would be necessary to transport a JSON Schema encoded in YAML. Storing it locally in YAML and developing in YAML makes sense, but transporting in YAML doesn't makes sense.

One of the goal is to support schema development and catalogs. Filed in #18 

> @Relequestual is going to reach out to the community to see if an `application/schema+yaml` media-type is something people actually want.

:+1: 

 
> **The `schema` media-type parameter**
> I'll suggest the necessary changes on the PR some time this week.

:+1: waitforit 

> **The `application/instance+schema` media type** We decided we do want to include this media type.

it's already in this PR, so :heavy_check_mark:

**Comment by richsalz** [comment]
Okay, question answered.  I'd like to see this brought up at the meeting and confirmed by the WG tho.

**Comment by jdesrosiers** [comment]
> Who is using YAML. Is there a need to standardize it _now_ ?

@Relequestual is going to reach out to the community to see if an `application/schema+yaml` media-type is something people actually want/need.

**Comment by ioggstream** [comment]
> > Who is using YAML. Is there a need to standardize it _now_ ?
> 
> @Relequestual is going to reach out to the community to see if an `application/schema+yaml` media-type is something people actually want/need.

Thanks! I filed #18 to track this YAML discussion even after this PR wilthout yaml will be merged :)

**Comment by stuartherbert** [comment]
JSON Schema _is_ written as YAML, when it is embedded in OpenAPI Spec v3.1 files. That would seem to pass the 'media types MUST NO be given names incorporating suffixes for structured syntaxes they do not actually employ" test in [RFC 6838](https://datatracker.ietf.org/doc/html/rfc6838#section-4.2.8)?

I personally don't know anyone extracting that schema and publishing it as a standalone resource _in YAML format_ today, mostly because OpenAPI Spec v3.1 adoption is still at an early phase (in the communities where I work). 

But I can certainly see it happening in the future. I have done work for clients who required schema definitions to be published as standalone, addressable resources. Once folks are used to writing schemas in YAML (as part of OpenAPI Spec v3.1), I do expect them to ask for any extracted schema to also be published in YAML for ease of comparison / consistency.

HTH.

**Comment by Relequestual** [comment]
> this thread will vanish once the PR will be merged

Do you mean closed as opposed to vanished?

**Comment by ioggstream** [comment]
Closed in favor of #19

**Pull request closed without merging**